chainstate/tx-verifier/src/transaction_verifier/input_output_policy.rs

```diff
@@ -159,40 +165,139 @@ pub fn check_tx_inputs_outputs_purposes(
+        | (TxOutput::ProduceBlockFromStake(..), TxOutput::DecommissionPool(..)) => true,
+        | (TxOutput::DecommissionPool(..), TxOutput::Transfer(..))
+        | (TxOutput::DecommissionPool(..), TxOutput::LockThenTransfer(..))
+        | (TxOutput::DecommissionPool(..), TxOutput::Burn(..))
+        | (TxOutput::DecommissionPool(..), TxOutput::CreateStakePool(..))
+        | (TxOutput::DecommissionPool(..), TxOutput::CreateDelegationId(..))
+        | (TxOutput::DecommissionPool(..), TxOutput::DelegateStaking(..)) => true,
+        | (TxOutput::DecommissionPool(..), TxOutput::ProduceBlockFromStake(..))
+        | (TxOutput::DecommissionPool(..), TxOutput::SpendShareFromDelegation(..))
+        | (TxOutput::DecommissionPool(..), TxOutput::DecommissionPool(..)) => false,
+        | (TxOutput::CreateDelegationId(..), _) => false,
+        | (TxOutput::DelegateStaking(..), TxOutput::Transfer(..))
+        | (TxOutput::DelegateStaking(..), TxOutput::LockThenTransfer(..))
+        | (TxOutput::DelegateStaking(..), TxOutput::Burn(..))
+        | (TxOutput::DelegateStaking(..), TxOutput::CreateStakePool(..))
+        | (TxOutput::DelegateStaking(..), TxOutput::ProduceBlockFromStake(..))
+        | (TxOutput::DelegateStaking(..), TxOutput::DecommissionPool(..))
+        | (TxOutput::DelegateStaking(..), TxOutput::CreateDelegationId(..)) => false,
+        | (TxOutput::DelegateStaking(..), TxOutput::DelegateStaking(..))
+        | (TxOutput::DelegateStaking(..), TxOutput::SpendShareFromDelegation(..)) => true,
+        | (TxOutput::SpendShareFromDelegation(..), TxOutput::Transfer(..))
+        | (TxOutput::SpendShareFromDelegation(..), TxOutput::LockThenTransfer(..))
+        | (TxOutput::SpendShareFromDelegation(..), TxOutput::Burn(..))
+        | (TxOutput::SpendShareFromDelegation(..), TxOutput::CreateStakePool(..))
+        | (TxOutput::SpendShareFromDelegation(..), TxOutput::CreateDelegationId(..))
+        | (TxOutput::SpendShareFromDelegation(..), TxOutput::DelegateStaking(..)) => true,
+        | (TxOutput::SpendShareFromDelegation(..), TxOutput::ProduceBlockFromStake(..))
+        | (TxOutput::SpendShareFromDelegation(..), TxOutput::DecommissionPool(..))
+        | (TxOutput::SpendShareFromDelegation(..), TxOutput::SpendShareFromDelegation(..)) => false,
     }
 }
 
+fn is_valid_one_to_any_combination_for_tx(
+    input_utxo: &TxOutput,
+    outputs: &[TxOutput],
+) -> Result<(), ConnectTransactionError> {
+    if !is_delegation_spending(input_utxo, outputs) {
+        let valid_inputs = are_inputs_valid_for_tx(std::slice::from_ref(input_utxo));
+        ensure!(valid_inputs, ConnectTransactionError::InvalidInputTypeInTx);
+        let valid_outputs = are_outputs_valid_for_tx(outputs);
+        ensure!(
+            valid_outputs,
+            ConnectTransactionError::InvalidOutputTypeInTx
+        );
+    }
+    Ok(())
+}
+
+// single DelegateStaking input; zero or one DelegateStakingOutput + any number of SpendShareFromDelegation
```

Comment:
Why do we allow zero? That basically burns the bridge, unless the balance there is zero.

Reply:
... balance is zero + pool is decommissioned.

Reply:
this allows tx like `DelegateStaking` -> `LockThenTransfer` to completely spend the share without change

Reply:
What happens if people attempt to burn the bridge. Meaning, they attempt to spend without the DelegateStaking output while they still have balance?